test/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_service_test.clj

```diff
@@ -385,3 +385,21 @@
        (is (logged?
             #"shutdown-on-error triggered because of exception"
             :error))))))
+
+(deftest ^:integration overflow-test
+  (let [watch-dir (fs/temp-dir "overflow")
+        results (atom [])
+        callback (make-callback results)]
+    (with-app-with-config
+     app watch-service-and-deps {}
+     (let [service (tk-app/get-service app :FilesystemWatchService)]
+       (watch! service watch-dir callback))
+     ;; From experimenting the buffer seems to fill at 512 events.
+     ;; However the number of events varies by operating system.
+     (let [files (for [n (range 1000)]
+                   (fs/file watch-dir  (str n ".txt")))]
+       (doseq [f files]
+         (spit f "file content"))
+      (let [events #{{:type :unknown
+                      :path nil}}]
+        (is (= events (wait-for-events results events))))))))
```

Comment:
I thought that as well, but from my experiments on Linux it appears that we do not.

We will get normal events until we overflow the buffer at which point we get *an* overflow event that begins incrementing a counter of how many *additional* events occurred until we reset the key.

Here is the log output from running this test at trace:
```
2016-08-01 11:28:56,303 INFO  [p.t.s.w.filesystem-watch-core] Got 1 event(s) for watched-path /tmp/overflow1470076135277-2767012267
2016-08-01 11:28:56,304 DEBUG [p.t.s.w.filesystem-watch-core] Events:
({:type :unknown, :path ""})

2016-08-01 11:28:56,306 TRACE [p.t.s.w.filesystem-watch-core] orig-events:
({:context nil,
  :count 1488,
  :kind
  #object[java.nio.file.StandardWatchEventKinds$StdWatchEventKind 0x49c745cd "OVERFLOW"]})
```